src/glog/logging.h.in

```diff
@@ -109,6 +109,50 @@ typedef unsigned __int64 uint64;
 
 @ac_google_end_namespace@
 
+#include <functional>
+struct LogMessageTime {
+    explicit LogMessageTime (const struct::tm& time_struct_,
+                             const time_t& timestamp_, const int32_t& usecs_):
+        time_struct(time_struct_), ts(timestamp_), usecs(usecs_) {}
+
+    const time_t& timestamp() const { return ts; }
+    const int& sec() const { return time_struct.tm_sec; }
+    const int32_t& usec() const { return usecs; }
+    const int& min() const { return time_struct.tm_min; }
+    const int& hour() const { return time_struct.tm_hour; }
+    const int& day() const { return time_struct.tm_mday; }
+    const int& month() const { return time_struct.tm_mon; }
+    const int& year() const { return time_struct.tm_year; }
+    const int& dayOfWeek() const { return time_struct.tm_wday; }
+    const int& dayInYear() const { return time_struct.tm_yday; }
+    const int& dst() const { return time_struct.tm_isdst; }
+
+    private:
+        const struct::tm& time_struct;
+        const time_t& ts;
+        const int32_t& usecs;
+};
+
+struct LogMessageInfo {
+    explicit LogMessageInfo(const std::string& severity_,
+                            const std::string& filename_,
+                            const int& line_number_,
+                            const int& thread_id_,
+                            const LogMessageTime& time_):
+        severity(severity_), filename(filename_), line_number(line_number_),
+        thread_id(thread_id_), time(time_)
+    {}
+
+    const std::string &severity;
+    const std::string &filename;
+    const int &line_number;
+    const int &thread_id;
+    const LogMessageTime& time;
+};
+
```

Comment:
Perhaps the indent level should be two spaces here? (Also in an unnamed namespace below)

According to [Google C++ Style Guide](https://google.github.io/styleguide/cppguide.html#Spaces_vs._Tabs):
> Use only spaces, and indent 2 spaces at a time.